jfix-stdlib-concurrency/src/test/kotlin/ru/fix/stdlib/concurrency/futures/PendingFutureLimiterKTest.kt

```diff
@@ -0,0 +1,125 @@
+package ru.fix.stdlib.concurrency.futures
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertTimeoutPreemptively
+import ru.fix.dynamic.property.api.DynamicProperty
+import java.time.Duration
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.withLock
+
+private val assertTimeoutDuration = Duration.ofSeconds(10)
+private val internalLargeTimeoutMs: Long = TimeUnit.MINUTES.toMillis(1)
+private const val CHECKS_QUANTITY = 100
+
+internal class PendingFutureLimiterKTest {
+
+    private val limiter = PendingFutureLimiter(
+            DynamicProperty.of(2),
+            DynamicProperty.of(internalLargeTimeoutMs)
+    )
+
+
+    @Test
+    fun `waitAll with timeout WHEN enqueue completed future AND breaks exist THEN limiter doesn't stuck`() {
+        var checkCounter = 0
+        assertTimeoutPreemptively(assertTimeoutDuration, { "failed on $checkCounter check" }) {
+            while (checkCounter < CHECKS_QUANTITY) {
```

Comment:
It doesn't make much sense to use 'while' construction having counter iterating loop. 'for' should be more suitable.

Reply:
after verbal conversation i removed this test-class, replaced its test-functions with single expressive test in PendingFutureLimiterTest.java